documentation/bash_cheatsheet.md

```diff
@@ -0,0 +1,215 @@
+# A Flash of Bash
+
+Git Town is written in Bash.
+Bash is a surprisingly powerful, versatile, and elaborate scripting system
+where every command-line application
+becomes a keyword in your programming language.
+
+With the right amount of structure,
+Bash programs can be as easily written and read
+as programs in many other languages.
+
+Bash is the
+[9th most popular language on GitHub](http://githut.info),
+and is actively used in most open-source repositories.
+
+Learning Bash not only allows you to contribute to Git Town,
+it will make your life easier by giving you the ability to
+create a plethora of useful little command-line helpers,
+the UNIX way!
+
+* [Shebang](#shebang)
+* [If clauses](#if-clauses)
+* [Functions](#functions)
+* [Arrays](#arrays)
+* [String Lists](#string-lists)
+* [Converting](#converting)
+
+
+## Shebang
+
+Use the cross-platform version of
+[shebang](https://en.wikipedia.org/wiki/Shebang_%28Unix%29):
+```bash
+#!/usr/bin/env bash
+```
+
+
+## If clauses
+
+* structure
+
+  ```bash
+  if [ condition ]; then
+    # code
+  elif [ condition ]; then
+    # code
+  else
+    # code
+  fi
+  ```
+
+* negation
+
+  ```bash
+  if [ ! condition ]; then
+    # runs if the condition is not true
```

Comment:
Can we remove this line too? I like your short one-line format that you used below.